### PR TITLE
nixos/rtorrent: add systemd hardening configuration

### PR DESCRIPTION
**Pull request by thiagokokada**
## Description of changes

Add systemd configuration to harden `rtorrent` service.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by thiagokokada — COMMENTED** [review]
CC @mweinelt since you helped me in PR #317530.

**Comment by thiagokokada** [comment]
@ofborg test rtorrent

File: nixos/tests/rtorrent.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  name = "rtorrent";
+  meta = {
+    maintainers = with pkgs.lib.maintainers; [ thiagokokada ];
+  };
+
+  nodes.machine = { pkgs, ... }: {
+    services.rtorrent = {
+      inherit port;
+      enable = true;
+      openFirewall = true;
```
Comment (mweinelt) [inline nixos/tests/rtorrent.nix:15]: redundant for connections on localhost

Comment (thiagokokada) [reply nixos/tests/rtorrent.nix]: Removed.

**Comment by thiagokokada** [comment]
@ofborg test rtorrent

**Comment by thiagokokada** [comment]
Any other comment here @mweinelt?

**Comment by mweinelt** [comment]
Not an rtorrent user, so not really.

File: nixos/modules/services/torrent/rtorrent.nix [hunk 2]
```diff
@@ -203,6 +203,30 @@ in {
             ExecStart="${cfg.package}/bin/rtorrent -n -o system.daemon.set=true -o import=${rtorrentConfigFile}";
             RuntimeDirectory = "rtorrent";
             RuntimeDirectoryMode = 755;
```
Comment (mweinelt) [inline nixos/modules/services/torrent/rtorrent.nix:205]:
```suggestion
            RuntimeDirectoryMode = 750;
```

Maybe not world-readable? And add a UMask.

**Review by mweinelt — COMMENTED** [review]
Except for maybe this

File: nixos/modules/services/torrent/rtorrent.nix (hunk 2 above)
Comment (thiagokokada) [reply nixos/modules/services/torrent/rtorrent.nix]: Done.

Umask is already set in `rtorrent` itself: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/torrent/rtorrent.nix#L164.

**Comment by thiagokokada** [comment]
@ofborg test rtorrent

File: nixos/modules/services/torrent/rtorrent.nix [hunk 3]
```diff
@@ -202,7 +204,31 @@ in {
             ExecStartPre=''${pkgs.bash}/bin/bash -c "if test -e ${cfg.dataDir}/session/rtorrent.lock && test -z $(${pkgs.procps}/bin/pidof rtorrent); then rm -f ${cfg.dataDir}/session/rtorrent.lock; fi"'';
```
Comment (AndersonTorres) [inline nixos/modules/services/torrent/rtorrent.nix:204]: Unrelated, but it should use `getExe` in both bash and pidof.

**Review by AndersonTorres — APPROVED** [review]
Looks fine to me!

**Pull request merged**